src/librustc/mir/interpret/allocation.rs

```diff
@@ -694,6 +694,12 @@ impl<Tag, Extra> Allocation<Tag, Extra> {
     }
 }
 
+impl AllocationDefinedness {
+    pub fn all_bytes_undef(&self) -> bool {
+        self.initial == false && self.ranges.len() == 1
```

Comment:
if the `ranges.len()` is not `1`, don't we have to iterate to check the other ranges?

Reply:
The `ranges` are run-length encoded so if `ranges.len() > 1` then the second is a range of defined bytes.

Reply:
Could you move this up one block, i.e. right below `struct AllocationDefinedness`?

Reply:
Also please add a comment here explaining what you said in the GH comment.

And what about `len() == 0`?